includes/Shortcode/Stories_Shortcode.php

```diff
@@ -123,15 +132,42 @@ private function prepare_story_attrs( array $attributes ): array {
 	 *
 	 * @since 1.5.0
 	 *
-	 * @param array<string,string|int> $attributes Array of arguments for Story Query.
-	 * @return array<string,string|int> Array of story arguments to pass to Story_Query.
+	 * @param array<string,int|string> $attributes Array of arguments for Story Query.
+	 * @return array<string,mixed> Array of story arguments to pass to Story_Query.
 	 */
 	private function prepare_story_args( array $attributes ): array {
-		return [
+		$args = [
 			// Show 100 stories at most to avoid 500 errors.
 			'posts_per_page' => min( (int) $attributes['number_of_stories'], 100 ), // phpcs:ignore WordPress.WP.PostsPerPage.posts_per_page_posts_per_page
 			'order'          => 'ASC' === $attributes['order'] ? 'ASC' : 'DESC',
 			'orderby'        => $attributes['orderby'],
 		];
+
+		$taxonomies           = get_object_taxonomies( Story_Post_Type::POST_TYPE_SLUG );
+		$should_add_tax_query = false;
+
+		foreach ( $taxonomies as $taxonomy ) {
+			$should_add_tax_query = $should_add_tax_query || '' !== $attributes[ $taxonomy ];
```

Comment:
Nit: To avoid looping through all elements, we can break out early once the condition is reached.

```suggestion
			if ( '' !== $attributes[ $taxonomy ] ) {
				$should_add_tax_query = true;
				break;
			}
```